test/unit/lodash.spec.js

```diff
@@ -0,0 +1,568 @@
+                expect(_.trim.apply(_, params)).to.eql('some string to test function');
+            }
+        );
+    });
+
+    describe('trimEnd', function () {
+        before(function () {
+            params = datasets.trimEnd();
+        });
+        it('should match types of passing elements', function () {
+            expect(params[0]).to.be.a('string');
+            expect(params[1]).to.be.a('string');
+        });
+        it('should return trimmed string', function () {
+                expect(_.trimEnd.apply(_, params)).to.eql('some string to test');
+            }
+        );
+    });
+
+    describe('trimStart', function () {
+        before(function () {
+            params = datasets.trimStart();
+        });
+        it('should match types of passing elements', function () {
+            expect(params[0]).to.be.a('string');
+            expect(params[1]).to.be.a('string');
+        });
+        it('should return trimmed string', function () {
+                expect(_.trimStart.apply(_, params)).to.eql('some string to test function.....');
+            }
+        );
+    });
+
+    describe('truncate', function () {
+        before(function () {
+            params = datasets.truncate();
+        });
+        it('should match types of passing elements', function () {
+            expect(params[0]).to.be.a('string');
+            expect(params).to.have.length.within(1, 2);
+            if (params[1] !== undefined && params.length === 2) {
+                expect(params[1]).to.be.an('object');
+            }
+        });
+        it('should check if second argument have validate properties', function () {
+            var options = params[1];
+            if (options !== undefined && params.length === 2) {
+                expect(options).to.have.any.keys('length', 'omission', 'separator');
+            }
+            // powinien miec max 3 wlasciowsci
+            // pierwsza number
+        });
+        it('should return truncated string', function () {
+                expect(_.truncate.apply(_, params)).to.eql('some very long text...');
+            }
+        );
+    });
+
+    describe('unescape', function () {
+        before(function () {
+            params = datasets.unescape();
+        });
+        it('should match types of passing elements', function () {
+            expect(params[0]).to.be.a('string');
+        });
+        it('should return transformed string', function () {
+                expect(_.unescape.apply(_, params)).to.eql('tom & jerry were old friends');
+            }
+        );
+    });
+
+    describe('upperCase', function () {
+        before(function () {
+            params = datasets.upperCase();
+        });
+        it('should match types of passing elements', function () {
+            expect(params[0]).to.be.a('string');
+        });
+        it('should return transformed string', function () {
+                expect(_.upperCase.apply(_, params)).to.eql('SOME STRING TO TEST FUNCTION');
```

Comment:
In my opinion, without looking to this file, that solution is impossible to reveal